webhooks/pod_webhook.go

```diff
@@ -230,6 +257,146 @@ func (m *PodMutator) injectSidecar(
+		return err
+	}
+	m.Log.Info("deploying the kube-flagd-proxy service")
+	if err := m.Client.Create(ctx, newFlagdKubeProxyServiceManifest()); err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+	return nil
+}
+
+func newFlagdKubeProxyServiceManifest() *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kubeProxyServiceName,
+			Namespace: currentNamespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{
+				"app.kubernetes.io/name": kubeProxyDeploymentName,
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Name:       "kube-flagd-proxy",
+					Port:       kubeProxyPort,
+					TargetPort: intstr.FromInt(kubeProxyPort),
+				},
+			},
+		},
+	}
+}
+
+func newFlagdKubeProxyManifest() *appsV1.Deployment {
+	replicas := int32(1)
+	return &appsV1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kubeProxyDeploymentName,
+			Namespace: currentNamespace,
+			Labels: map[string]string{
+				"app": kubeProxyDeploymentName,
+			},
+		},
+		Spec: appsV1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": kubeProxyDeploymentName,
+				},
+			},
+
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app":                    kubeProxyDeploymentName,
+						"app.kubernetes.io/name": kubeProxyDeploymentName,
+					},
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: kubeProxyServiceAccountName,
+					Containers: []corev1.Container{
+						{
+							Image: fmt.Sprintf("%s:%s", kubeProxyImage, kubeProxyTag),
+							Name:  kubeProxyDeploymentName,
+							Ports: []corev1.ContainerPort{
+								{
+									Name:          "port",
+									ContainerPort: kubeProxyPort,
+								},
+							},
+							Args: []string{
+								"start",
+								"--debug",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (m *PodMutator) handleKubeProxy(ctx context.Context, sidecar *corev1.Container, source v1alpha1.Source) error {
+	// does the proxy exist
+	exists, ready, err := m.isKubeProxyReady(ctx)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		//create, defer pod deployment
+		if err = m.deployKubeProxy(ctx); err != nil { // error is for some reason still deploying the pod which is wrong
+			return fmt.Errorf("unable to deploy flagd-kube-proxy: %w", err)
+		}
+		return &kubeProxyDeferError{}
+	}
+	if exists && !ready {
+		return &kubeProxyDeferError{}
+	}
+
+	config := []types.SourceConfig{
+		{
+			Provider: "grpc",
+			Selector: fmt.Sprintf("core.openfeature.dev/%s", source.Source),
+			URI:      fmt.Sprintf("grpc://%s.%s.svc.cluster.local:%d", kubeProxyServiceName, currentNamespace, kubeProxyPort),
+		},
+	}
+	configB, err := json.Marshal(config)
+	if err != nil {
+		return err
+	}
+
+	sidecar.Args = append(
+		sidecar.Args,
+		"--sources",
+		string(configB),
+		"--debug",
```

Comment:
Should debugging be enabled by default?

Reply:
@james-milligan do you plan to introduce this through helm [1] ? 

I think this is an essential configuration - to enable/disable debug logs for flagd sidecar & kube-proxy 

[1] - https://github.com/open-feature/open-feature-operator/blob/main/chart/open-feature-operator/values.yaml#L4

Reply:
yes, all configurations will follow the existing flow, using environment variable overrides for the default values, and then relevant configurations being overwritten though custom resources

IMO I don't believe there are any configurations in this implementation that require configurations via custom resources, all can be handled through the helm chart / env vars

Reply:
ive introduced this configuration, along with port, metrics port and image/tag

Reply:
> I think this is an essential configuration - to enable/disable debug logs for flagd sidecar & kube-proxy

adding this configuration to the sidecar requires changes to the FalgSourceConfiguration CRD, which is outside of the scope of this PR, ive opened an issue here: https://github.com/open-feature/open-feature-operator/issues/416

Reply:
Perfect thanks :)